controllers/bookController.js

```diff
@@ -94,3 +95,47 @@ exports.deleteBook = catchAsync(async (req, res, next) => {
     data: null,
   });
 });
+exports.borrowBook = catchAsync(async (req, res, next) => {
+  const id = req.params.id;
+  const book = await Book.findById(id);
+  if (!book) {
+    return next(new AppError("No book found with that ID", 404));
+  }
+  if (book.available === false) {
```

Comment:
Why not do this?

```suggestion
  if (!book.available) {
```
That is a better representation